### ISSUE TITLE
past court dates should appear with casa_case details

### ISSUE THREAD

**Issue by Zrrrpy** [rubyforgood/casa#805]
Part of epic #880 // dependent on #807 

**What type of user is this for? [ volunteer/supervisor/admin/all or All CASA Admin]**
all

**Where does/should this occur?**
On the Edit CASA Case page (example: https://casa-qa.herokuapp.com/casa_cases/2/edit)
![Screen Shot 2020-09-26 at 4 30 47 PM (2)](https://user-images.githubusercontent.com/62810851/94352459-149e4b00-001a-11eb-92be-e9c7c29195e0.png)

**Description**
All users should see all past `court_dates` associated with a `casa_case` on the Edit CASA Case page.
All `court_dates` that have passed should become `past_court_dates` and appear here. 

**Backend**
This is part of a new feature, so an addition will need to be made to db for `past_court_dates` which can be associated with a `casa_case`.

**Comment by littleforest** [comment]
See #808 for new table structure discussion.

**Comment by compwron** [comment]
Because this was unassigned, I moved it back to "to do" on the board

**Comment by matisnape** [comment]
Hey @compwron can I take this over? 
Also, how do I get access to https://casa-qa.herokuapp.com/casa_cases/2/edit ? Or is it just an example to show the route to view?

**Comment by jmkoni** [comment]
@matisnape assigned! it is just an example to show the route, but you can access QA with the same un/password combos that are in the readme.

**Comment by matisnape** [comment]
Hey, I'm setting up the project from master branch, but I've encountered several problems:
1. With `bin/rails spec` I get 2 fails 
![image](https://user-images.githubusercontent.com/2719923/95373611-88bfd500-08dd-11eb-982d-0044a1890cb9.png)

2. `yarn test` also fails 
![image](https://user-images.githubusercontent.com/2719923/95374075-2c10ea00-08de-11eb-9523-3bde60076fd9.png)

3. It also looks that master has code that was not formatted - when I run linters, several files have been updated, ERB linter says that several files have too long lines and cannot be auto-corrected and standardrb mentions useless assignment in `spec/system/admin_views_volunteers_page_spec.rb:134:7`

Please advise what I should do with the above - fix them or ignore them?

**Comment by compwron** [comment]
@matisnape The linters are not part of CI right now to reduce friction for contributors. Feel free to make a PR to fix lint! In general, try to keep lint changes in a separate PR (or at least a separate commit) from non lint changes. 

As for the rails spec failures... that is strange. The build is passing on CI so it must be some local issue but I'm not sure what it could be. If you're around for office hours tonight 6-8pm Pacific we can pair on it then. :)

`yarn spec` is still experimental and it is not in `.travis.yml` (i.e the CI build) yet, so don't worry about that right now.

**Comment by matisnape** [comment]
@compwron 
Thanks for info :)
I think first I will do the work for this issue and in case of problems with those tests in the PR, we will pair. How about that?

**Comment by matisnape** [comment]
@compwron I am a bit confused about the acceptance criteria here and the fact that there are some related things described in other issues. 
Below are my questions:
1) DB for `court_date` - I see that `casa_case` model has field for `court_date`. Considering that this issue mentions past dates, I understand that a new model for court dates is needed and it will be related to casa_case with has_many association. Is that correct?
2) If 1) is confirmed, I assume you would like it to be displayed somewhere here (see screenshot), correct? Or have a separate table?
![image](https://user-images.githubusercontent.com/2719923/95662406-04688e80-0b37-11eb-99d5-ff1b192fd13e.png)

**Comment by compwron** [comment]
@matisnape 
1. yes
2. Yes that is a good place to show past court dates for a case.

**Comment by compwron** [comment]
@matisnape how is it going? :)

**Comment by matisnape** [comment]
@compwron sorry, I missed your previous comment! I will work on this during the weekend :) I'm Ruby for Good's Slack, so in case I have additional questions, I will post there :)

**Comment by matisnape** [comment]
@compwron business question: how should the court date CRUD be implemented? 

This is how I see it:
1) Create court date instance when new casa case is created - court date form is included in new casa case view
2) After CC is created, new court dates can be added separately only - so with separate forms just for court dates
3) I would display the court dates as a separate table where dates can be edited/deleted

Could you povide your insight on the matter?

**Comment by Zrrrpy** [comment]
Hey @matisnape! 

Thank you so much for all your work on this 👏  

Only one "next court date" will ever exist at once, so admins/supervisors will not need the ability to add a new court date until _after_ the next court date has passed (#881 provides more context on this). Once it has passed, that date should appear in a new section labeled "Past Court Dates" – this is the piece this issue seeks to address. 

The area you highlighted above would work great for placement of past court dates. Please be mindful that some users may also be displaying this data on mobile web.

**Comment by matisnape** [comment]
@Zrrrpy Why the past court dates should appear on Edit casa case, and not on Show?

**Comment by Zrrrpy** [comment]
hey @matisnape – thanks so much for all your work on this!! So exciting to see a PR up. 👏 You can expect it to be reviewed by a Lead Dev in the next day or so. 

Past court dates certainly can display on the "Show" CASA Case page instead of on the "Edit" CASA Case page, as long as it is done in a way that does not overwhelm the page viewer, especially with all their `case_contacts` displaying on the same page. (Some `casa_cases` may acquire upwards of 40 or more past court dates.) If you have a vision for this then go for it!
